File: sdk/android/story-sdk/src/main/kotlin/com/openstory/sdk/OpenStoryConfiguration.kt
package com.openstory.sdk

data class OpenStoryConfiguration(
    val clientId: String,
    val staticToken: String,
    val baseUrl: String,
    val connectTimeoutMillis: Long = 5_000L,
    val readTimeoutMillis: Long = 10_000L,
) {
    init {
        require(clientId.isNotBlank()) { "clientId must not be blank." }
        require(staticToken.isNotBlank()) { "staticToken must not be blank." }
        require(baseUrl.isNotBlank()) { "baseUrl must not be blank." }
        require(connectTimeoutMillis > 0) { "connectTimeoutMillis must be positive." }
        require(readTimeoutMillis > 0) { "readTimeoutMillis must be positive." }
    }

    internal fun normalizedBaseUrl(): String = baseUrl.trim().trimEnd('/')
}

File: sdk/flutter/lib/src/models/open_story_configuration.dart
class OpenStoryConfiguration {
  const OpenStoryConfiguration({
    required this.clientId,
    required this.staticToken,
    required this.baseUrl,
    this.connectTimeoutMillis = 5000,
    this.readTimeoutMillis = 10000,
  }) : assert(clientId != ""),
       assert(staticToken != ""),
       assert(baseUrl != ""),
       assert(connectTimeoutMillis > 0),
       assert(readTimeoutMillis > 0);

  final String clientId;
  final String staticToken;
  final String baseUrl;
  final int connectTimeoutMillis;
  final int readTimeoutMillis;

  Map<String, Object> toMap() {
    return <String, Object>{
      "clientId": clientId,
      "staticToken": staticToken,
      "baseUrl": baseUrl,
      "connectTimeoutMillis": connectTimeoutMillis,
      "readTimeoutMillis": readTimeoutMillis,
    };
  }
}
